geode-lucene/src/main/java/org/apache/geode/cache/lucene/LuceneQuery.java

```diff
@@ -37,21 +42,38 @@
  * Results are returned in order of their score with respect to this query.
  * </p>
  */
-@Experimental
 public interface LuceneQuery<K, V> {
   /**
    * Execute the query and return the region keys that match this query, up to the limit specified
    * by {@link #getLimit()}.
-   * 
+   *
+   * @return Collection of Apache Geode region keys that satisfy the Lucene query.
    * @throws LuceneQueryException if the query could not be parsed or executed.
+   * @throws CacheClosedException if the cache was closed while the Lucene query was being executed.
+   * @throws PrimaryBucketException if the primary bucket was moved from the node on which the
```

Comment:
PrimaryBucketException is an internal exception, which should not be part of the API. Do we actually throw this exception from this method? It seems like we should be retrying in any case.